composer/metrics/nlp.py

```diff
@@ -692,15 +792,19 @@ def update(self, batch: Dict[str, Any], outputs: List[str], labels: List[str]):
         del labels  # never used
         client = self.get_client()
 
+        metric_result_dict = copy.deepcopy(self.metric_result_dict)
```

Comment:
why deepcopy now? same as above

Reply:
the class variable is a template for each new batch to construct what should be logged. It's the same thing we do here: https://github.com/mosaicml/composer/blob/dev/composer/datasets/in_context_learning_evaluation.py#L618